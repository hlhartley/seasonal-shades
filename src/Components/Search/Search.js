import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const Search = (props) => {
    if (props[props.type + 'Colors'].length) {
        return(
            <div className='color-search'>
                <p><b>Total # {props.type} colors:</b> <u>{`${props[props.type + 'Colors'].length}`}</u> results</p>
                <div>
                    <i className="fas fa-search"></i>
                    <input type="text" onChange={props.handleChange} placeholder='Search by color name' className='search-input'></input>
                </div>
            </div>
        )
    } else {
        return(
            <div className='color-search'>
                <p><b>Total # {props.type} colors:</b> <u>0</u> results</p>
                <div>
                    <i className="fas fa-search"></i>
                    <input type="text" onChange={props.handleChange} placeholder='Search by color name' className='search-input'></input>
                </div>
            </div>
        )
    }
};

export const mapStateToProps = (state) => ({
    nail_polishColors: state.nailpolishColors,
    lipstickColors: state.lipstickColors,
    blushColors: state.blushColors,
    eyeshadowColors: state.eyeshadowColors,
});

Search.propTypes = {
    handleChange: PropTypes.func,
    type: PropTypes.string,
    nail_polishColors: PropTypes.array,
    lipstickColors: PropTypes.array,
    blushColors: PropTypes.array,
    eyeshadowColors: PropTypes.array,
}

export default connect(mapStateToProps)(Search)
