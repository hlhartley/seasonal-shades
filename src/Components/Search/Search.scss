.color-search {
    align-items: center;
    background-color: $tan-color;
    display: flex;
    flex-flow: column;
    justify-content: center;
    margin-top: -15px;
    padding-bottom: 20px;
    padding-top: 20px;
}

.fa-search {
    font-size: 20px;
    margin-right: 10px;
}

.search-input {
    padding: 5px 5px 5px 10px;
    width: 200px;
}