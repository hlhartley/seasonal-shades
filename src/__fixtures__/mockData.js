export const allColors = {
    522: {
        product: {},
        hexcode: '#EED8BE'
    },
    771: {
        product: {},
        hexcode: '#BD9E9B' 
    },
        'casablanca': {
        product: {},
        hexcode: '#444446'
    }
};

export const mockState = {
    allColors: {
        522: {
            product: {},
            hexcode: '#EED8BE'
        },
        771: {
            product: {},
            hexcode: '#BD9E9B' 
        },
        'casablanca': {
            product: {},
            hexcode: '#444446'
        }
    },
    favorites: {
        '001lune': true,
        '001petal': true,
    },
    blushColors: undefined,
    eyeshadowColors: undefined,
    lipstickColors: [],
    nail_polishColors: undefined,
    moreColors: undefined,
};

export const mockFavorites = {
    '001lune': true,
    '001petal': true, 
}