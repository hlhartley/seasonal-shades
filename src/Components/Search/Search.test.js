import React from 'react';
import Search, { mapStateToProps } from './Search';
import { shallow } from 'enzyme';
import { mockState } from '../../__fixtures__/mockData';

describe('Search', () => {
    describe('Search Component', () => {
        let wrapper;
        let mockHandleChange;
    
        beforeEach(() => {
            mockHandleChange = jest.fn()
            wrapper = shallow(<Search handleChange={mockHandleChange} lipstickColors={[]} blushColors={undefined} eyeshadowColors={undefined} nail_polishColors={undefined}/>)
        });
    
        it('should take a snapshot and pass in the data correctly', () => {
            expect(wrapper).toMatchSnapshot()
        });
    });

    describe('mapStateToProps', () => {
        const expected = {
            blushColors: undefined,
            eyeshadowColors: undefined,
            lipstickColors: [],
            nail_polishColors: undefined,
        }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
    });
});