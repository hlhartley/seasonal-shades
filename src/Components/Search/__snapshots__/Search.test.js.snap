// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search Component should take a snapshot and pass in the data correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
